src/sentry/coreapi.py

```diff
@@ -335,10 +343,10 @@ def _process_data_timestamp(self, data, current_datetime=None):
             current_datetime = datetime.now()
 
         if value > current_datetime + timedelta(minutes=1):
-            raise InvalidTimestamp('Invalid value for timestamp (in future): %r' % value)
+            raise FutureInvalidTimestamp('%s' % (value - current_datetime))
 
         if value < current_datetime - timedelta(days=30):
-            raise InvalidTimestamp('Invalid value for timestamp (too old): %r' % value)
+            raise PastInvalidTimestamp('%s' % (value - current_datetime))
```

Comment:
Yeah, so I'm open to ideas here. I initially thought to just use `total_seconds()`, but it wasn't as readable on larger values. At least superficially. Ideally, this would be interpreted by the frontend and converted to a nice timestamp. But.... hey. Lemme see if there's somethign better I can do here in python.